new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSystemResOverrides.java

```diff
@@ -0,0 +1,514 @@
+          managedServerPodNamePrefix + i, domainNamespace);
+      checkPodReady(managedServerPodNamePrefix + i, domainUid, domainNamespace);
+    }
+  }
+
+  //deploy application sitconfig.war to domain
+  private void deployApplication(String targets) {
+    logger.info("Getting node port for T3 channel");
+    int t3channelNodePort = assertDoesNotThrow(()
+        -> getServiceNodePort(domainNamespace, adminServerPodName + "-external", "t3channel"),
+        "Getting admin server t3channel node port failed");
+    assertNotEquals(-1, t3ChannelPort, "admin server t3channelport is not valid");
+
+    //deploy application
+    logger.info("Deploying webapp {0} to domain", sitconfigAppPath);
+    deployUsingWlst(K8S_NODEPORT_HOST, Integer.toString(t3channelNodePort),
+        ADMIN_USERNAME_DEFAULT, ADMIN_PASSWORD_DEFAULT, targets, sitconfigAppPath,
+        domainNamespace);
+  }
+
+  /**
+   * Create a WebLogic domain on a persistent volume by doing the following. Create a configmap containing WLST script
+   * and property file. Create a Kubernetes job to create domain on persistent volume.
+   *
+   * @param wlstScriptFile python script to create domain
+   * @param domainPropertiesFile properties file containing domain configuration
+   * @param pvName name of the persistent volume to create domain in
+   * @param pvcName name of the persistent volume claim
+   * @param namespace name of the domain namespace in which the job is created
+   */
+  private void createDomainOnPVUsingWlst(Path wlstScriptFile, Path domainPropertiesFile,
+      String pvName, String pvcName, String namespace) {
+    logger.info("Preparing to run create domain job using WLST");
+
+    List<Path> domainScriptFiles = new ArrayList<>();
+    domainScriptFiles.add(wlstScriptFile);
+    domainScriptFiles.add(domainPropertiesFile);
+
+    logger.info("Creating a config map to hold domain creation scripts");
+    String domainScriptConfigMapName = "create-domain-scripts-cm";
+
+    assertDoesNotThrow(
+        () -> createConfigMapForDomainCreation(
+            domainScriptConfigMapName, domainScriptFiles, namespace, this.getClass().getSimpleName()),
+        "Create configmap for domain creation failed");
+
+    // create a V1Container with specific scripts and properties for creating domain
+    V1Container jobCreationContainer = new V1Container()
+        .addCommandItem("/bin/sh")
+        .addArgsItem("/u01/oracle/oracle_common/common/bin/wlst.sh")
+        .addArgsItem("/u01/weblogic/" + wlstScriptFile.getFileName()) //wlst.sh
+        // script
+        .addArgsItem("-skipWLSModuleScanning")
+        .addArgsItem("-loadProperties")
+        .addArgsItem("/u01/weblogic/" + domainPropertiesFile.getFileName());
+    //domain property file
+
+    logger.info("Running a Kubernetes job to create the domain");
+    createDomainJob(image, isUseSecret, pvName, pvcName, domainScriptConfigMapName,
+        namespace, jobCreationContainer);
+  }
+
+  /**
```

Comment:
Is this not a common test utility ?

Reply:
I am calling createDockerRegistrySecret in this method when secret doesn't exist. But createOCRRepoSecret itself is not a common utility.